### PR TITLE
*/*: Remove redundant `|| die` guards

### PR DESCRIPTION
**Pull request by SoapGentoo**
* Since all ebuilds in the tree are EAPI>=4,
  `|| die` on builtin commands are redundant
  and dead code.

Signed-off-by: David Seifert <soap@gentoo.org>

### CODE REVIEW

File: app-emulation/containers-storage/containers-storage-1.15.1.ebuild [hunk 1]
```diff
@@ -56,5 +56,5 @@ src_install() {
 }
 
 src_test() {
-	GOPATH="${S}" unshare -m emake local-test-unit || die
+	GOPATH="${S}" unshare -m emake local-test-unit
```
Comment (SoapGentoo) [inline app-emulation/containers-storage/containers-storage-1.15.1.ebuild]: false positive

File: app-emulation/containers-storage/containers-storage-1.15.2.ebuild [hunk 2]
```diff
@@ -56,5 +56,5 @@ src_install() {
 }
 
 src_test() {
-	GOPATH="${S}" unshare -m emake local-test-unit || die
+	GOPATH="${S}" unshare -m emake local-test-unit
```
Comment (SoapGentoo) [inline app-emulation/containers-storage/containers-storage-1.15.2.ebuild]: false positive

File: app-mobilephone/kannel-sqlbox/kannel-sqlbox-1.5.0-r1.ebuild [hunk 3]
```diff
@@ -49,15 +49,14 @@ src_configure() {
...
 }
 
 src_install() {
-	emake DESTDIR="${D}" install || die "failed emake install"
+	emake DESTDIR="${D}" install
 
 	if use doc; then
-		emake doc/userguide.html || die "emake docs failed"
+		emake doc/userguide.html
 		dohtml doc/userguide.html || die "userguide.html not found"
```
Comment (ulm) [inline app-mobilephone/kannel-sqlbox/kannel-sqlbox-1.5.0-r1.ebuild]: Missed one.

File: dev-erlang/protobuffs/protobuffs-0.9.0.ebuild [hunk 4]
```diff
@@ -31,7 +31,7 @@ DOCS=( AUTHORS  ChangeLog  README.md RELNOTES.md )
 S="${WORKDIR}/${MY_P}"
 
 src_test() {
-	./scripts/generate_emakefile.escript || die
+	./scripts/generate_emakefile.escript
```
Comment (ulm) [inline dev-erlang/protobuffs/protobuffs-0.9.0.ebuild]: False positive?

File: dev-java/jamvm/jamvm-2.0.0-r1.ebuild [hunk 5]
```diff
@@ -86,7 +86,7 @@ src_install() {
 	dosym /usr/bin/jamvm ${JDK_DIR}/bin/java
 	for files in ${CLASSPATH_DIR}/g*; do
 		if [ $files = "${CLASSPATH_DIR}/bin/gjdoc" ] ; then
-			dosym $files ${JDK_DIR}/bin/javadoc || die
+			dosym $files ${JDK_DIR}/bin/javadoc
 		else
 			dosym $files \
 				${JDK_DIR}/bin/$(echo $files|sed "s#$(dirname $files)/g##") || die
```
Comment (ulm) [inline dev-java/jamvm/jamvm-2.0.0-r1.ebuild]: Missed one.

File: dev-java/sbt-bin/sbt-bin-1.1.4.ebuild [hunk 6]
```diff
@@ -45,5 +45,5 @@ src_install() {
 	doins -r conf || die
```
Comment (ulm) [inline dev-java/sbt-bin/sbt-bin-1.1.4.ebuild]: Missed  one.

File: dev-java/sbt-bin/sbt-bin-1.2.8.ebuild [hunk 7]
```diff
@@ -45,5 +45,5 @@ src_install() {
 	doins -r conf || die
```
Comment (ulm) [inline dev-java/sbt-bin/sbt-bin-1.2.8.ebuild]: .. and another.

File: dev-java/sbt-bin/sbt-bin-1.3.2.ebuild [hunk 8]
```diff
@@ -45,5 +45,5 @@ src_install() {
 	doins -r conf || die
```
Comment (ulm) [inline dev-java/sbt-bin/sbt-bin-1.3.2.ebuild]: ... and yet another.

File: dev-java/sbt-bin/sbt-bin-1.1.5.ebuild [hunk 9]
```diff
@@ -45,5 +45,5 @@ src_install() {
 	doins -r conf || die
 
 	fperms 0755 "${dest}/lib/sbt" || die
```
Comment (ulm) [inline dev-java/sbt-bin/sbt-bin-1.1.5.ebuild]: Another two, `doins` and `fperms`.

File: dev-lang/parrot/parrot-7.1.0-r1.ebuild [hunk 10]
```diff
@@ -55,20 +55,20 @@ src_configure() {
...
 
 src_test() {
-	emake -j1 test || die
+	emake -j1 test
 }
 
 src_install() {
-	emake -j1 install-dev DESTDIR="${D}" DOC_DIR="${EPREFIX}/usr/share/doc/${PF}" || die
+	emake -j1 install-dev DESTDIR="${D}" DOC_DIR="${EPREFIX}/usr/share/doc/${PF}"
 	dodoc CREDITS DONORS.pod PBC_COMPAT PLATFORMS RESPONSIBLE_PARTIES TODO || die
```
Comment (ulm) [inline dev-lang/parrot/parrot-7.1.0-r1.ebuild]: Missed one.

File: dev-lang/parrot/parrot-7.1.0-r1.ebuild [hunk 11]
```diff
@@ -55,20 +55,20 @@ src_configure() {
...
-	dosym parrot-ops2c /usr/bin/ops2c || die
+	dosym parrot-ops2c /usr/bin/ops2c
 	if use examples; then
 		insinto "/usr/share/doc/${PF}/examples"
 		doins -r examples/* || die
```
Comment (ulm) [inline dev-lang/parrot/parrot-7.1.0-r1.ebuild]: Missed one.

File: dev-lang/parrot/parrot-7.11.0-r1.ebuild [hunk 12]
```diff
@@ -55,20 +55,20 @@ src_configure() {
...
 
 src_install() {
-	emake -j1 install-dev DESTDIR="${D}" DOC_DIR="${EPREFIX}/usr/share/doc/${PF}" || die
+	emake -j1 install-dev DESTDIR="${D}" DOC_DIR="${EPREFIX}/usr/share/doc/${PF}"
 	dodoc CREDITS DONORS.pod PBC_COMPAT PLATFORMS RESPONSIBLE_PARTIES TODO || die
-	dosym parrot-ops2c /usr/bin/ops2c || die
+	dosym parrot-ops2c /usr/bin/ops2c
 	if use examples; then
 		insinto "/usr/share/doc/${PF}/examples"
 		doins -r examples/* || die
```
Comment (ulm) [inline dev-lang/parrot/parrot-7.11.0-r1.ebuild]: Missed one.

File: sci-biology/mira/mira-4.0.2.ebuild [hunk 13]
```diff
@@ -81,7 +81,7 @@ src_configure() {
 #src_compile() {
 #	base_src_compile
 #	# TODO: resolve docbook incompatibility for building docs
-#	if use doc; then emake -C doc clean docs || die; fi
+#	if use doc; then emake -C doc clean docs
```
Comment (SoapGentoo) [inline sci-biology/mira/mira-4.0.2.ebuild]: broken (commented)

File: net-print/cndrvcups-common-lb/cndrvcups-common-lb-3.70.ebuild [hunk 14]
```diff
@@ -80,7 +80,7 @@ src_compile() {
 	change_dir emake
 
 	# Cannot be moved to 'change_dir' as it doesn't need eautoreconf
-	( cd "${S}/c3plmod_ipc" && emake ) || die
+	( cd "${S}/c3plmod_ipc" && emake )
```
Comment (SoapGentoo) [inline net-print/cndrvcups-common-lb/cndrvcups-common-lb-3.70.ebuild]: broken

File: net-print/cndrvcups-common-lb/cndrvcups-common-lb-3.60-r1.ebuild [hunk 15]
```diff
@@ -77,7 +77,7 @@ src_compile() {
 	change_dir emake
 
 	# Cannot be moved to 'change_dir' as it doesn't need eautoreconf
-	( cd "${S}/c3plmod_ipc" && emake ) || die
+	( cd "${S}/c3plmod_ipc" && emake )
```
Comment (SoapGentoo) [inline net-print/cndrvcups-common-lb/cndrvcups-common-lb-3.60-r1.ebuild]: broken

File: net-libs/netwib/netwib-5.39.0.ebuild [hunk 16]
```diff
@@ -40,7 +40,7 @@ src_prepare() {
 }
 
 src_configure() {
-	sh genemake || die
```
Comment (SoapGentoo) [inline net-libs/netwib/netwib-5.39.0.ebuild]: broken

File: media-gfx/iscan/iscan-2.30.4.2.ebuild [hunk 17]
```diff
@@ -126,7 +126,7 @@ src_install() {
 		done
 		if [[ "/plug-ins" != "${gimpplugindir}" ]]; then
 			dodir ${gimpplugindir}
-			dosym $(realpath -s --relative-to="${gimpplugindir}" "${EPREFIX}/usr/bin/iscan" || die) "${gimpplugindir}"/iscan
```
Comment (SoapGentoo) [inline media-gfx/iscan/iscan-2.30.4.2.ebuild]: broken

File: mate-base/caja/caja-1.22.1.ebuild [hunk 18]
```diff
@@ -78,7 +78,7 @@ src_test() {
 	unset SESSION_MANAGER
 	unset DBUS_SESSION_BUS_ADDRESS
 
-	Xemake check || die "Test phase failed"
+	Xemake check
```
Comment (SoapGentoo) [inline mate-base/caja/caja-1.22.1.ebuild]: broken

File: dev-lang/pcc/pcc-1.1.0-r1.ebuild [hunk 19]
```diff
@@ -29,9 +29,9 @@ src_configure() {
 }
 
 src_compile() {
-	emake  || die "emake failed"
+	emake
```
Comment (SoapGentoo) [inline dev-lang/pcc/pcc-1.1.0-r1.ebuild]: broken

File: dev-lang/parrot/parrot-7.11.0-r1.ebuild (hunk 12 above)
Comment (SoapGentoo) [reply dev-lang/parrot/parrot-7.11.0-r1.ebuild]: will do that in a followup PR

File: dev-lang/mlton/mlton-20130715.ebuild [hunk 20]
```diff
@@ -101,8 +101,8 @@ src_install() {
 		pax-mark m "${S}/usr/bin/mlyacc"
 		mv "${S}/usr" "${D}" || die "mv failed"
 	else
-		emake DESTDIR="${D}" install-no-docs || die
-		if use doc; then emake DESTDIR="${D}" TDOC="${D}"/usr/share/doc/${P} install-docs || die; fi
```
Comment (SoapGentoo) [inline dev-lang/mlton/mlton-20130715.ebuild]: broken

File: dev-java/jamvm/jamvm-2.0.0-r1.ebuild [hunk 21]
```diff
@@ -86,7 +86,7 @@ src_install() {
 	dosym /usr/bin/jamvm ${JDK_DIR}/bin/java
 	for files in ${CLASSPATH_DIR}/g*; do
 		if [ $files = "${CLASSPATH_DIR}/bin/gjdoc" ] ; then
-			dosym $files ${JDK_DIR}/bin/javadoc || die
+			dosym $files ${JDK_DIR}/bin/javadoc
 		else
 			dosym $files \
```
Comment (SoapGentoo) [inline dev-java/jamvm/jamvm-2.0.0-r1.ebuild]: broken

File: dev-erlang/protobuffs/protobuffs-0.9.0.ebuild [hunk 22]
```diff
@@ -31,7 +31,7 @@ DOCS=( AUTHORS  ChangeLog  README.md RELNOTES.md )
 S="${WORKDIR}/${MY_P}"
 
 src_test() {
-	./scripts/generate_emakefile.escript || die
```
Comment (SoapGentoo) [inline dev-erlang/protobuffs/protobuffs-0.9.0.ebuild]: broken

File: x11-themes/larry-backgrounds/larry-backgrounds-20110822-r1.ebuild [hunk 23]
```diff
@@ -63,7 +63,7 @@ src_install() {
 	local share_home='backgrounds/larry-the-cow'
 
 	insinto /usr/share/${share_home}/
-	( cd "${DISTDIR}" && doins ${A} ) || die
+	cd "${DISTDIR}" && doins ${A}
```
Comment (ulm) [inline x11-themes/larry-backgrounds/larry-backgrounds-20110822-r1.ebuild]: What happens if `cd` fails?

File: dev-libs/mpfr/mpfr-3.1.3_p4.ebuild [hunk 24]
```diff
@@ -49,7 +49,7 @@ multilib_src_install_all() {
 
 	# clean up html/license install
 	pushd "${ED}"/usr/share/doc/${PF} >/dev/null
-	dohtml *.html && rm COPYING* *.html || die
+	dohtml *.html && rm COPYING* *.html
```
Comment (ulm) [inline dev-libs/mpfr/mpfr-3.1.3_p4.ebuild]: False positive.

File: dev-libs/mpfr/mpfr-3.1.4.ebuild [hunk 25]
```diff
@@ -50,6 +50,6 @@ multilib_src_install_all() {
 
 	# clean up html/license install
 	pushd "${ED}"/usr/share/doc/${PF} >/dev/null
-	dohtml *.html && rm COPYING* *.html || die
+	dohtml *.html && rm COPYING* *.html
```
Comment (ulm) [inline dev-libs/mpfr/mpfr-3.1.4.ebuild]: False positive.

File: sci-mathematics/singular/singular-4.0.2.ebuild [hunk 26]
```diff
@@ -92,8 +92,8 @@ src_compile() {
 # 	dodoc README
 # 	# execs and libraries
 # 	cd "${S}"/build/bin
-# 	dobin ${MY_PN}* gen_test change_cost solve_IP toric_ideal LLL \
-# 		|| die "failed to install binaries"
+# 	dobin ${MY_PN}* gen_test change_cost solve_IP toric_ideal LLL
+#
```
Comment (ulm) [inline sci-mathematics/singular/singular-4.0.2.ebuild]: This introduces an empty line.

File: dev-ruby/rubygems/rubygems-2.7.10.ebuild [hunk 27]
```diff
@@ -95,7 +95,7 @@ each_ruby_install() {
 	insinto "${sld#${EPREFIX}}"  # bug #320813
 	newins "${FILESDIR}/auto_gem.rb.ruby19" auto_gem.rb || die
 
-	newbin bin/gem $(basename ${RUBY} | sed -e 's:ruby:gem:') || die
+	newbin bin/gem $(basename ${RUBY} | sed -e 's:ruby:gem:')
```
Comment (ulm) [inline dev-ruby/rubygems/rubygems-2.7.10.ebuild]: The `die` should be moved into the parenthesis, to catch `sed` failures?
This applies to all ebuilds of this package.

File: net-libs/axtls/axtls-1.5.3.ebuild [hunk 28]
```diff
@@ -117,7 +117,7 @@ use_flag_config() {
 			config/.config || die
 	fi
 
-	yes "n" | emake -j1 oldconfig > /dev/null || die
+	yes "n" | emake -j1 oldconfig > /dev/null
```
Comment (mgorny) [inline net-libs/axtls/axtls-1.5.3.ebuild]:
```suggestion
	emake -j1 oldconfig < <(yes n) > /dev/null
```

File: net-libs/axtls/axtls-1.5.3.ebuild [hunk 29]
```diff
@@ -137,7 +137,7 @@ multilib_src_configure() {
 			ewarn "No saved config, seeding with the default"
 			cp "${FILESDIR}"/config config/.config || die
 		fi
-		yes "" | emake -j1 oldconfig > /dev/null || die
+		yes "" | emake -j1 oldconfig > /dev/null
```
Comment (mgorny) [inline net-libs/axtls/axtls-1.5.3.ebuild]:
```suggestion
		emake -j1 oldconfig < <(yes '') > /dev/null
```

File: net-libs/axtls/axtls-2.1.3.ebuild [hunk 30]
```diff
@@ -119,7 +119,7 @@ use_flag_config() {
 			config/.config || die
 	fi
 
-	yes "n" | emake -j1 oldconfig > /dev/null || die
+	yes "n" | emake -j1 oldconfig > /dev/null
```
Comment (mgorny) [inline net-libs/axtls/axtls-2.1.3.ebuild]: Also in this ebuild.

File: net-libs/axtls/axtls-2.1.4.ebuild [hunk 31]
```diff
@@ -119,7 +119,7 @@ use_flag_config() {
 			config/.config || die
 	fi
 
-	yes "n" | emake -j1 oldconfig > /dev/null || die
+	yes "n" | emake -j1 oldconfig > /dev/null
```
Comment (mgorny) [inline net-libs/axtls/axtls-2.1.4.ebuild]: Here too.

File: net-libs/axtls/axtls-2.1.5.ebuild [hunk 32]
```diff
@@ -119,7 +119,7 @@ use_flag_config() {
 			config/.config || die
 	fi
 
-	yes "n" | emake -j1 oldconfig > /dev/null || die
+	yes "n" | emake -j1 oldconfig > /dev/null
```
Comment (mgorny) [inline net-libs/axtls/axtls-2.1.5.ebuild]: Guess.

**Pull request closed without merging**